org.eclipse.buildship.core.test/src/main/groovy/org/eclipse/buildship/core/configuration/internal/BuildConfigurationPersistenceTest.groovy

```diff
@@ -0,0 +1,152 @@
+package org.eclipse.buildship.core.configuration.internal
+
+import spock.lang.IgnoreIf
+import spock.lang.Shared
+import spock.lang.Subject
+import spock.util.environment.OperatingSystem
+
+import com.gradleware.tooling.toolingclient.GradleDistribution
+
+import org.eclipse.core.resources.IProject
+import org.eclipse.core.resources.IResource
+import org.eclipse.core.resources.ProjectScope
+import org.eclipse.core.runtime.NullProgressMonitor
+
+import org.eclipse.buildship.core.CorePlugin
+import org.eclipse.buildship.core.configuration.ProjectConfiguration
+import org.eclipse.buildship.core.test.fixtures.WorkspaceSpecification
+
+class BuildConfigurationPersistenceTest extends WorkspaceSpecification {
+
+    @Shared
+    @Subject
+    BuildConfigurationPersistence persistence = new BuildConfigurationPersistence()
+
+    IProject project
+
+    void setup() {
+        project = newProject("sample-project")
+    }
+
+    def "can't save null as preferences"() {
+        when:
+        persistence.saveBuildConfiguration(project, null)
+
+        then:
+        thrown NullPointerException
+    }
+
+    def "can save preferences"() {
+        setup:
+        persistence.saveBuildConfiguration(project, buildConfigProperties())
+
+        expect:
+        def node = new ProjectScope(project).getNode(CorePlugin.PLUGIN_ID)
+        node.get(BuildConfigurationPersistence.PREF_KEY_CONNECTION_GRADLE_DISTRIBUTION, null) == 'GRADLE_DISTRIBUTION(WRAPPER)'
+    }
+
+    def "read validates input"() {
+        when:
+        persistence.readBuildConfiguratonProperties(null)
+
+        then:
+        thrown NullPointerException
+
+        // TODO (donat) should we do the validation in the persistence or in the Manager?
```

Comment:
At the level that knows whether something is wrong. So here I'd say in the persistence.